server/master/src/main/java/org/apache/accumulo/master/Master.java

```diff
@@ -1336,7 +1332,11 @@ public void assignedTablet(KeyExtent extent) {
   }
 
   @Override
-  public Collection<KeyExtent> migrations() {
-    return migrations.keySet();
+  public Set<KeyExtent> migrations() {
+    Set<KeyExtent> migrationsCopy = new HashSet<KeyExtent>();
+    synchronized (migrations) {
+      migrationsCopy.addAll(migrations.keySet());
+    }
+    return migrationsCopy;
```

Comment:
>  I was thinking a bigger risk would be that a user would accidentally introduce a RuntimeException 

My hope was that the runtime exception would possibly expose a bug (in addition to causing a bug).  However, there are certainly cases where it would only cause problems and not expose any bugs.

> wrapped immutability is not something which we can check at compile-time

That would be nice to have.